rest-api/etl/etl.sql

```diff
@@ -767,58 +716,47 @@ CREATE TABLE cdm.src_person_location
     zip varchar(255),
     state_ppi_code varchar(255),
     state varchar(255),
+    location_id bigint,
     PRIMARY KEY (participant_id)
 );
 
-DROP VIEW IF EXISTS cdm.temp_src_person_location;
-CREATE VIEW cdm.temp_src_person_location AS
-SELECT
-    src_m.participant_id,
-    src_m.question_ppi_code,
-    src_m.value_string,
-    IF(src_m.question_ppi_code = 'PIIAddress_StreetAddress',
-                src_m.value_string, NULL)                          AS address_1,
-    IF(src_m.question_ppi_code = 'PIIAddress_StreetAddress2',
-                src_m.value_string, NULL)                          AS address_2,
-    IF(src_m.question_ppi_code = 'StreetAddress_PIICity',
-                src_m.value_string, NULL)                          AS city,
-    IF(src_m.question_ppi_code = 'StreetAddress_PIIZIP',
-                src_m.value_string, NULL)                          AS zip,
-    IF(src_m.question_ppi_code = 'StreetAddress_PIIState' AND src_m.topic_value = 'States',
-                src_m.value_ppi_code, NULL)                        AS state_ppi_code
-FROM cdm.src_mapped src_m
-INNER JOIN cdm.src_participant src_p
-    ON  src_m.participant_id = src_p.participant_id
-    AND src_m.date_of_survey = src_p.latest_date_of_location
-WHERE
-    src_m.question_ppi_code IN (
-        'PIIAddress_StreetAddress',
-        'PIIAddress_StreetAddress2',
-        'StreetAddress_PIICity',
-        'StreetAddress_PIIZIP',
-        'StreetAddress_PIIState'
-    )
-;
-
 INSERT cdm.src_person_location
 SELECT
-    src.participant_id                          AS participant_id,
-    MAX(src.address_1)                          AS address_1,
-    MAX(src.address_2)                          AS address_2,
-    MAX(src.city)                               AS city,
-    MAX(src.zip)                                AS zip,
-    MAX(src.state_ppi_code)                     AS state_ppi_code,
-    MAX(RIGHT(src.state_ppi_code, 2))           AS state
-FROM cdm.temp_src_person_location src
-GROUP BY
-    src.participant_id
-HAVING
-    COUNT(DISTINCT src.address_1) <= 2
-    AND COUNT(DISTINCT src.address_2) <= 2
-    AND COUNT(DISTINCT src.city) <= 2
-    AND COUNT(DISTINCT src.zip) <= 2
-    AND COUNT(DISTINCT src.state_ppi_code) <= 2
-;
+    src_participant.participant_id        AS participant_id,
+    MAX(m_address_1.value_string)         AS address_1,
```

Comment:
Existing behavior, but I'm failing to understand why it would make sense to call `MAX` on a column named m_address_1. My only real concerned is the interaction with the DISTINCT address question below.

Reply:
MySQL has a mode (enabled by default since 5.7.5) called ONLY_FULL_GROUP_BY:

https://dev.mysql.com/doc/refman/5.7/en/sql-mode.html#sqlmode_only_full_group_by

which rejects queries with GROUP BY where other columns referenced aren't aggregated in some way.

So even if we expect there will be exactly one row in the results, we still need to aggregate the other columns.